src/karma.config.js

```diff
@@ -0,0 +1,126 @@
+/* global process */
+
+let chromeFlags = [];
+
+process.env.CHROME_BIN = require('puppeteer').executablePath();
+
+// In Docker, the tests run as root, so the sandbox must be disabled.
+if (process.env.RUNNING_IN_DOCKER) {
+  chromeFlags.push('--no-sandbox');
+
+  // Disable `/dev/shm` usage as this can cause Chrome to fail to load large
+  // HTML pages, such as the one Karma creates with all the tests loaded.
+  //
+  // See https://github.com/GoogleChrome/puppeteer/issues/1834 and
+  // https://github.com/karma-runner/karma-chrome-launcher/issues/198.
+  chromeFlags.push('--disable-dev-shm-usage');
+
+  // Use Chromium from Alpine packages. The one that Puppeteer downloads won't
+  // load in Alpine.
+  process.env.CHROME_BIN = 'chromium-browser';
+}
+
+module.exports = function (config) {
+  let testFiles = ['**/test/*-test.js'];
+
+  config.set({
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: './',
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['browserify', 'mocha', 'chai', 'sinon'],
+
+    // list of files / patterns to load in the browser
+    files: [
+      // Test setup
+      '../test/bootstrap.js',
+
+      // Test modules.
+      ...testFiles.map(pattern => ({
+        pattern,
+
+        // Disable watching because karma-browserify handles this.
+        watched: false,
+
+        type: 'js',
+      })),
+    ],
+
+    // list of files to exclude
+    exclude: [],
+
+    // preprocess matching files before serving them to the browser
+    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+    preprocessors: {
+      '../test/bootstrap.js': ['browserify'],
+      '**/*-test.js': ['browserify'],
+    },
+
+    browserify: {
+      debug: true,
+      transform: [
+        [
+          'babelify',
+          {
+            // The existence of this preset option is due to a config issue with where jsx modules
+            // are not transpiled to js.
+            // See https://github.com/hypothesis/client/issues/2929
+            presets: require('../.babelrc.cjs').presets,
+            extensions: ['.js'],
+          },
+        ],
+      ],
+    },
+
+    mochaReporter: {
+      // Display a helpful diff when comparing complex objects
+      // See https://www.npmjs.com/package/karma-mocha-reporter#showdiff
+      showDiff: true,
+      // Output only summary and errors in development to make output easier to parse.
+      output: 'minimal',
+    },
+
+    // Use https://www.npmjs.com/package/karma-mocha-reporter
+    // for more helpful rendering of test failures
+    reporters: ['progress', 'mocha'],
+
+    // web server port
+    port: 9876,
```

Comment:
Might be useful in future to pick a different port that doesn't conflict with the client's tests. Let's leave as-is for the moment.

Reply:
@robertknight , the client and lms both use this port too, we probably want 3 different ports.